chronos/asyncsync.nim

```diff
@@ -267,16 +267,32 @@ proc addFirstNoWait*[T](aq: AsyncQueue[T], item: T) {.
   aq.addFirstImpl(item)
 
 proc addLastNoWait*[T](aq: AsyncQueue[T], item: T) {.
-    raises: [AsyncQueueFullError].}=
+     raises: [AsyncQueueFullError].} =
   ## Put an item ``item`` at the end of the queue ``aq`` immediately.
   ##
   ## If queue ``aq`` is full, then ``AsyncQueueFullError`` exception raised.
   if aq.full():
     raise newException(AsyncQueueFullError, "AsyncQueue is full!")
   aq.addLastImpl(item)
 
+proc addFirstNoWaitSafe*[T](aq: AsyncQueue[T], item: T) {.
```

Comment:
how about a `distinct UnboundedQueue` for this case?

Reply:
I'm not sure i understand, i need AsyncQueue in that place, i do not need any other queues.

Reply:
the idea is to not have these `Safe` suffixes but instead use a separate type for the unbounded use case - "safe" here is easy to interpret as "never raises defect" when in fact it is the opposite: the caller can only call them in special cases where it is safe to do so.

Reply:
The problem is that AsyncQueue i'm using is not unbounded. It is bounded and has limits, but in the way i'm using it - it is safe.

Reply:
should we call it `addFirstNotFull` then?